custom/enikshay/reports/filters.py

```diff
@@ -32,6 +32,12 @@ def selected(self):
         :return: Selected locations without their descendants
         """
         location_ids = self.request.GET.getlist(self.slug)
+
+        if len(location_ids) == 0 \
+                and not self.request.couch_user.has_permission(self.request.domain, 'access_all_locations'):
+            # Display the user's location in the filter if none is selected
+            location_ids = self.request.couch_user.get_location_ids(self.request.domain)
+
```

Comment:
Nice, I like this idea.  `request.can_access_all_locations` is also available as long as it's a real request (not like, a test mock)